conformance_tests/test_tiovx/test_copy_swap.c

```diff
@@ -0,0 +1,1513 @@
+    {
+        parameter = vxGetParameterByIndex(node, 0);
+        EXPECT_NE_VX_STATUS(VX_FAILURE, status = vxGetStatus((vx_reference)parameter));
+
+		if (VX_SUCCESS == status)
+        {
+            vxReleaseParameter(&parameter);
+            status = VX_FAILURE;
+        }
+        else if (new_num_nodes < original_num_nodes)
+        {
+            status = VX_SUCCESS;
+        }
+    }
+    return status;
+}
+
+/*  Test copy node optimisation
+    An input copy node should be disconnected if its output is virtual and can't be modified
+    An input copy node won't be removed if it's output can be modified
+    An output copy node should be disconnected if its input is virtual and not connected to another input
+    An output copy node won't be removed if its input is connected to another input
+    An embedded copy node should be removed if the input is not connected to another input
+    An embedded copy node won't be removed is the input is connected to another input
+*/
+TEST(copySwap, testCopyRemoval)
+{
+    vx_status status = VX_SUCCESS;
+    vx_context context = context_->vx_context_;
+    vx_graph graph = NULL;
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+
+    int i;
+    vx_image images[] =
+    {
+        vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateVirtualImage(graph, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateVirtualImage(graph, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateVirtualImage(graph, 10, 10, VX_DF_IMAGE_U8),
+        vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8)
+    };
+
+    /* Create a graph with an input copy node with a fan-out of 2*/
+    vx_node nodes[] =
+    {
+        vxCopyNode(graph, (vx_reference)images[0], (vx_reference)images[1]),
+        vxNotNode(graph, images[1], images[2]),
+        vxNotNode(graph, images[1], images[3]),
+        NULL, NULL
+    };
+    writeImage(images[0], 23, 45);
+    addParameterToGraph(graph, nodes[0], 0);
+
+    /* Simple input copy node is removed */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, nodeIsOptimised(graph, nodes[0]));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    /* Graph executes correctly after input Copy node removal */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImage(images[2], 255-23, 255-45) | checkImage(images[3], 255-23, 255-45));
+
+    writeImage(images[6], 12, 34);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxSetGraphParameterByIndex(graph, 0, (vx_reference)images[6]));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    /* Graph executes correctly after vxSetGraphParameterByIndex to set input parameter */
+    status = checkImage(images[2], 255-12, 255-34); //-->issue here!
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, status);
+    status = checkImage(images[3], 255-12, 255-34);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, status);
+
+	vxReleaseNode(&nodes[0]);
```

Comment:
Minor, but looks like some whitespace issues in a few places similar to this

Reply:
done all replaces